### PR TITLE
Frontend & performance updates

### PR DESCRIPTION
**Pull request by jmazouri**
### Frontend
- Improve performance of promotions page by making comments only load when a campaign is expanded
- Improve / centralize resolving of discord channel, user, and URL tags 
- Move Infractions to Logs page
- Add Redirect route type (currently to redirect `/infractions` to `/logs/infractions`)
- Added `isButton ` property to `ModixRoute.ts` - if set, the route will show up on the right side of the navbar (the title should be an emoji)
- Add optional claims to routes - these are claims that the user needs to have at least one of (instead of all of) for the route to show up in navigation. This is automatically populated with the route's children's required claims.
- Stopped checking for login twice on load by removing the check from the root App component
- Set devtool to source-map in `vue.config.js` to allow debugging from vscode
- Add `tslint.json` with `"defaultSeverity"` set to off to stop VSCode from yelling (we can figure out standards later)
- Various minor layout improvements, and improvements to cases where table filtering/sorting wasn't working

### Backend
- `DeletedMessageSummary` sort fields are now named by the actual property rather than the navigation property
- Deleting a Ban infraction will not try to remove the ban if the infraction was already rescinded
- Removed explicit `"identify"` scope from `ModixAuthenticationExtensions` because it comes implicitly from `DiscordAuthenticationOptions`
- Added `ModixUser.FromSocketGuildUser`, used in `AutocompleteController`
- Retrieve avatar URLs from discord.net instead of OAuth

Reviewer Protip: hide whitespace changes.

### CODE REVIEW

File: Modix/ClientApp/src/components/Logs/Tabs.vue [hunk 1]
```diff
@@ -1,74 +1,89 @@
...
-            if (!route.routeData.requiredClaims)
-            {
-                return "Disallowed";
-            }
+    mounted()
+    {
 
-            return "Required Claims: " + route.routeData.requiredClaims.toString();
-        }
     }
```
[the remark this answers is not in the document]
Comment (calledude) [reply Modix/ClientApp/src/components/Logs/Tabs.vue]: Jmaz might have force-pushed this, I can't find it anymore

[the remark this answers is not in the document]
Comment (Scott-Caldwell) [reply Modix/ClientApp/src/components/Logs/Tabs.vue]: It's the `mounted()` function

[the remark this answers is not in the document]
Comment (calledude) [reply Modix/ClientApp/src/components/Logs/Tabs.vue]: Right, I see it now, my bad

[the remark this answers is not in the document]
Comment (jmazouri) [reply Modix/ClientApp/src/components/Logs/Tabs.vue]: Oh, yeah, that's unnecessary. Removed.

File: Modix/ClientApp/src/models/User.ts [hunk 2]
```diff
@@ -8,6 +8,6 @@ export default class User
 
     get avatarUrl(): string
```
[the remark this answers is not in the document]
Comment (jmazouri) [reply Modix/ClientApp/src/models/User.ts]: Prooobably not, I just forgot to refactor. Fixed.

**Comment by jrusbatch** [comment]
I'm not looking

**Pull request merged**